### PR TITLE
style: remove unused code in tests

### PR DESCRIPTION
**Pull request by faddat**
Closes: #XXX

## What is the purpose of the change

This PR removes unused code in tests.

## Brief Changelog

* remove unused constants, variables, and functions in osmosis' tests.

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: tests/ibc-hooks/ibc_middleware_test.go [hunk 1]
```diff
@@ -48,9 +48,6 @@ type HooksTestSuite struct {
 	pathAB *ibctesting.Path
 	pathAC *ibctesting.Path
 	pathBC *ibctesting.Path
-	pathBA *ibctesting.Path
-	pathCA *ibctesting.Path
-	pathCB *ibctesting.Path
```
Comment (pysel) [inline tests/ibc-hooks/ibc_middleware_test.go:53]: imo it might make sense to leave these, just in case they might be useful in the future

**Review by pysel — APPROVED** [review]
lgtm!

File: tests/e2e/e2e_setup_test.go [hunk 2]
```diff
@@ -18,8 +18,6 @@ const (
 	skipUpgradeEnv = "OSMOSIS_E2E_SKIP_UPGRADE"
 	// Environment variable name to skip the IBC tests
 	skipIBCEnv = "OSMOSIS_E2E_SKIP_IBC"
-	// Environment variable name to skip state sync testing
-	skipStateSyncEnv = "OSMOSIS_E2E_SKIP_STATE_SYNC"
```
Comment (p0mvn) [inline tests/e2e/e2e_setup_test.go:22]: This is being used here, we should remove:
https://github.com/osmosis-labs/osmosis/blob/6e1b7a727b2a4ad5b44b962423494bb6826e91ca/tests/e2e/e2e_test.go#L1334

Comment (faddat) [reply tests/e2e/e2e_setup_test.go:22]: hmmmm I thought it was used.  Any idea how toget one's editor to see that?  Or ci for that matter?

Comment (p0mvn) [reply tests/e2e/e2e_setup_test.go]: Not going to block this PR on this change but we can replace the hardcoded value with `skipStateSyncEnv` in the future.

Thanks @faddat

**Review by p0mvn — APPROVED** [review]

**Pull request merged**